Cargo.toml

```diff
@@ -20,7 +20,8 @@ serde_json = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 slog = "2.7.0"
 wapc-guest = "0.4.0"
+serde_yaml = "0.8.23"
```

Comment:
Can't we leave that to be a dev dependency? It was a dev dependency just to allow some quick way inside of the tests to load a config definition

Reply:
we now need it for [this](https://github.com/kubewarden/policy-sdk-rust/pull/32/files/3b13c27e710c2407d509c3eda1a3f0e54dae957f#diff-7a94d8b8ce50e72e4f8aa8d35ba93292ce08d57fd0b604b7f7c121033b9054d4R86) which was moved from the [policy-fetcher](https://github.com/kubewarden/policy-fetcher/blob/main/src/verify/config.rs#L46)

Reply:
I do wonder if that `VerificationConfig` enum should expose a `serde_yaml::Value` in the `Invalid` variant.

I am not sure about the `Invalid` case to be honest. I wonder if we could refactor that so that there is no `Invalid` variant but rather a `Result<VerificationConfig>`.

Reply:
It could be done, but right now we use that to provide user output on why a config [is invalid](https://github.com/kubewarden/policy-fetcher/blob/75dfde5f6ecc619147646d55d48ab9ce84e1b5b8/src/verify/config.rs#L169-L190).

Reply:
I wonder under what circumstances `serde_yaml::from_str` unamrshals a `VerificationConfig::Invalid(_)`.

Reply:
An example could be:
```
---
apiVersion: v1
allOf:
  - kind: gitlabActionThisDoesntExistYet
    owner: kubewarden
anyOf: ~
```

It is only useful once one calls `build_latest_verification_config()`, which returns the serde errors:
```
$ kwctl verify --verification-config-path ./broken-config.yml registry://ghcr.io/kubewarden/tests/pod-privileged:v0.1.9
Error: Not a valid configuration file: Err(Message("unknown variant `gitlabActionThisDoesntExistYet`, expected one of `pubKey`, `genericIssuer`, `githubAction`", None))
```

Reply:
In that case I think this should be part of the `E` associated type in the `Err(E)` variant of the `Result`. Unless I'm misunderstanding something, I think `OK(T)` should be reserved to the happy path, and be as explicit as we need with the associated type `E` in order to describe what went wrong. Having an error description on the `OK(T)` side seems weird to me.

Reply:
In any case, not for this PR, sorry for the noise.